localstack/services/sns/publisher.py

```diff
@@ -0,0 +1,950 @@
+                    "statusCode": status_code,
+                    "providerResponse": payload.read(),
+                }
+                store_delivery_log(context.message, subscriber, success=True, delivery=delivery)
+
+        except Exception as exc:
+            LOG.info(
+                "Unable to run Lambda function on SNS message: %s %s", exc, traceback.format_exc()
+            )
+            store_delivery_log(context.message, subscriber, success=False)
+            message_body = create_sns_message_body(
+                message_context=context.message, subscriber=subscriber
+            )
+            sns_error_to_dead_letter_queue(subscriber, message_body, str(exc))
+
+    def prepare_message(self, message_context: SnsMessage, subscriber: SnsSubscription):
+        external_url = external_service_url("sns")
+        unsubscribe_url = create_unsubscribe_url(external_url, subscriber["SubscriptionArn"])
+        # see the format here https://docs.aws.amazon.com/lambda/latest/dg/with-sns.html
+        # issue with sdk to serialize the attribute inside lambda
+        message_attributes = prepare_message_attributes(message_context.message_attributes)
+        event = {
+            "Records": [
+                {
+                    "EventSource": "aws:sns",
+                    "EventVersion": "1.0",
+                    "EventSubscriptionArn": subscriber["SubscriptionArn"],
+                    "Sns": {
+                        "Type": message_context.type or "Notification",
+                        "MessageId": message_context.message_id,
+                        "TopicArn": subscriber["TopicArn"],
+                        "Subject": message_context.subject,
+                        "Message": message_context.message_content(subscriber["Protocol"]),
+                        "Timestamp": timestamp_millis(),
+                        "SignatureVersion": "1",
+                        # TODO Add a more sophisticated solution with an actual signature
+                        # Hardcoded
+                        "Signature": "EXAMPLEpH+..",
+                        "SigningCertUrl": "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-0000000000000000000000.pem",
+                        "UnsubscribeUrl": unsubscribe_url,
+                        "MessageAttributes": message_attributes,
+                    },
+                }
+            ]
+        }
+        return event
+
+
+class SqsTopicPublisher(BaseTopicPublisher):
+    def _publish(self, context: SnsPublishContext, subscriber: SnsSubscription):
+        message_context = context.message
+        try:
+            message_body = self.prepare_message(message_context, subscriber)
+            sqs_message_attrs = self.create_sqs_message_attributes(
+                subscriber, message_context.message_attributes
+            )
+        except Exception:
+            LOG.exception("An internal error occurred while trying to send the message to SQS")
```

Comment:
Again, not sure if this it within the scope of the PR. It would maybe make sense to add the message (or something else) to the log so we can identify the cause

Reply:
I'm using `LOG.exception` which adds the stack trace to the log message automatically at the logging level `ERROR`.

Reply:
Ah neat, missed that!